include/oneapi/dpl/internal/dynamic_selection_impl/sycl_backend.h

```diff
@@ -106,27 +141,45 @@ class sycl_backend
                       report_value_v<SelectionHandle, execution_info::task_time_t>)
         {
             std::chrono::steady_clock::time_point t0;
+            bool use_event_profiling = q.template has_property<sycl::property::queue::enable_profiling>();
             if constexpr (report_value_v<SelectionHandle, execution_info::task_time_t>)
             {
-                t0 = std::chrono::steady_clock::now();
+                if (!use_event_profiling)
+                {
+                    t0 = std::chrono::steady_clock::now();
+                }
             }
             auto e1 = f(q, std::forward<Args>(args)...);
-            auto e2 = q.submit([=](sycl::handler& h) {
-                h.depends_on(e1);
-                h.host_task([=]() {
-                    if constexpr (report_value_v<SelectionHandle, execution_info::task_time_t>)
-                        s.report(execution_info::task_time, (std::chrono::steady_clock::now() - t0).count());
-                    if constexpr (report_info_v<SelectionHandle, execution_info::task_completion_t>)
-                    {
+            if constexpr(report_info_v<SelectionHandle, execution_info::task_completion_t>){
+                auto e2 = q.submit([=](sycl::handler& h){
+                    h.depends_on(e1);
+                    h.host_task([=](){
                         s.report(execution_info::task_completion);
-                    }
+                    });
                 });
-            });
-            return async_waiter{e2};
+                return async_waiter{e2, new SelectionHandle(s)};
```

Comment:
I know we don't currently have a policy that requires reporting of both task completion and time, but this implementation would not support it because of this return. If we assumed they were mutually exclusive, the next condition would be an "else if", but its just an "if" so that both might be supported.

Reply:
Fixed